src/framework/global/types/translatablestring.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef MU_FRAMEWORK_TRANSLATABLESTRING_H
+#define MU_FRAMEWORK_TRANSLATABLESTRING_H
+
+#include "translation.h"
+
+#include "types/string.h"
+
+#ifndef NO_QT_SUPPORT
+#include <QString>
+#endif
+
+namespace mu {
+//! Note: in order to make the string visible for `lupdate`,
+//! you must write TranslatableString(...) explicitly.
+struct TranslatableString {
+    const char* context = nullptr;
+    const char* str = nullptr;
+    const char* disambiguation = nullptr;
+
+    TranslatableString() = default;
+    TranslatableString(const char* context, const char* str, const char* disambiguation = nullptr)
+        : context(context), str(str), disambiguation(disambiguation) {}
+
+    inline bool isValid() const
+    {
+        // Not null or empty
+        return context && context[0] && str && str[0];
```

Comment:
Do we ever want an empty translatable string? Perhaps this line should just be an assertion in the constructor?

Reply:
We certainly want the ability to default-construct a TranslatableString. For example, to use it as the default return value for a method, or for "optional" arguments for methods (i.e. if you pass nothing, an empty string will be passed).